firebase-firestore/ktx/src/androidTest/java/com/google/firebase/firestore/ComponentRegistrarTest.kt

```diff
@@ -0,0 +1,166 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore
+
+import com.google.common.truth.Truth.assertThat
+import com.google.firebase.Timestamp
+import com.google.firebase.firestore.ktx.annotations.KDocumentId
+import com.google.firebase.firestore.ktx.annotations.KServerTimestamp
+import com.google.firebase.firestore.testutil.testCollection
+import com.google.firebase.firestore.testutil.waitFor
+import java.util.Date
+import kotlinx.serialization.Contextual
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
+import org.junit.Test
+
+class ComponentRegistrarTest {
+
+    @Test
+    fun documentReference_is_using_ktx_mapencoder_when_set_to_firestore() {
```

Comment:
Just tested, the current Java POJO method,  `documentReference.set()` method will not crash with no-argument constructor, it can save the data to firestore, and we can actuall call `documentSnapshot.getData()` method to get the data back. 

It only crashes when we try to get the saved data back using `documentSnapshot.toObject()`

Another difference we can use to do this assertion is `@Exclude` annotation. Since we do not plan to change this Java annotation, this annotation is also transparent to Kotlin Serialization:
```
 @Serializable class Student( @get: Exclude val ifUSeeThisFieldKtxSerializationIsNotWorking:Boolean, val name:String)
 
 Java save it as: {name=foo}
 Kotlin save it as: {name=foo, ifUSeeThisFieldKtxSerializationIsNotWorking=true}
 ```